### DISCUSSION TITLE
Documenting API endpoints in /server/api

### DISCUSSION THREAD

**Discussion by justindasilva** [nuxt/nuxt#18296] [Questions]
So I want to make use of the new `/server/api` folder in Nuxt 3 (Nuxt Bridge).

Is there a standard way to document your endpoints and generate a documentation kinda like Swagger UI?

I've seen this type of annotations:
```
/**
 * @openapi
 * /api/items/getAll
 *   get:
 *     description: Return all items
 *     responses:
 *       200:
 *         description: Returns a list of items
 */
```

I would like to use this type of annotation ideally (JSDoc):
```
/**
 * An item
 * @typedef {object} Item
 * @property {string} name - Name of item
 */

/**
 * GET /api/items/getAll
 * @summary Return all items
 * @return {array<Item>} 200 - success response - application/json
 */
```

It would be cool if a module existed that would support the second annotation type above to document API endpoints and generate a UI (kinda like Swagger does) at some path like `/api/docs`

Perhaps someone can lead me in the right direction on how to accomplish this?

**Comment by pi0** [comment]
Yes! We have been working on [unjs/untyped](https://github.com/unjs/untyped) to make similar thing happen at some time :)

**Reply by justindasilva** [reply]
Cool, that looks awesome!

Are there any plans in the making for a Nuxt module that would make use of the `untyped` library? The idea being to recommend documenting your API endpoints and perhaps generate a UI very much like Swagger. Either that, or just convert what is defined to the OpenAPI format that Swagger expects.

If there are no plans in the making for something like this, I feel like I should give a shot to creating my first NPM Nuxt Module.
Are there any official docs on how to create your own Nuxt module?
